source/request_response.c

```diff
@@ -630,15 +630,24 @@ int aws_http_message_get_request_method(
     return aws_raise_error(error);
 }
 
+static struct aws_byte_cursor s_default_empty_path = AWS_BYTE_CUR_INIT_FROM_STRING_LITERAL("/");
+
 int aws_http_message_set_request_path(struct aws_http_message *request_message, struct aws_byte_cursor path) {
     AWS_PRECONDITION(request_message);
     AWS_PRECONDITION(aws_byte_cursor_is_valid(&path));
     AWS_PRECONDITION(request_message->request_data);
 
+    /* Its valid for uri path to be empty, but http spec requires empty paths to
+    be sent as "/", so default it here. */
+    if (path.len == 0) {
+        path = s_default_empty_path;
```

Comment:
You should change aws_http_message_get_request_path() to return "/" when the path isn't set yet.

It feels weird if we're autocorrecting empty paths to be "/" but we're no autocorrecting unset paths to be "/"

Reply:
agree. addressed in the next commit.